File: server/ruby/app/infrastructure/mappers/payment_details/payment_product5500_specific_output_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PaymentProduct5500SpecificOutputMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::PaymentProduct5500SpecificOutput.new(
            payment_reference: response.payment_reference,
            payment_end_date: response.payment_end_date,
            payment_start_date: response.payment_start_date,
            entity_id: response.entity_id
          )
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/payment_product_5500_specific_output_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.payment_product_5500_specific_output import PaymentProduct5500SpecificOutput
from onlinepayments.sdk.domain.payment_product5500_specific_output import PaymentProduct5500SpecificOutput as PaymentProduct5500SpecificOutputSdk

class PaymentProduct5500SpecificOutputMapper:

    @staticmethod
    def map_from_sdk_response(
        response: Optional[PaymentProduct5500SpecificOutputSdk],
    ) -> Optional[PaymentProduct5500SpecificOutput]:
        if response is None:
            return None

        return PaymentProduct5500SpecificOutput(
            payment_reference = response.payment_reference,
            payment_end_date = response.payment_end_date,
            payment_start_date = response.payment_start_date,
            entity_id = response.entity_id,
        )